documentation/jdisc/developing-osgi-bundles.html

```diff
@@ -141,9 +141,57 @@ <h2 id="typical-bundle">What is an OSGi bundle?</h2>
     </p>
 
 <h2 id="depending-on-JNI-libraries">Depending on JNI Libraries</h2>
-<p>
-To use JNI libraries, add the .so file(s) to a bundle and depend on this as any other dependency.
-Tensorflow is an example bundle with JNI.
-Read more on <a href="container-components.html#JNI-requires-restart">
-JNI and deployments</a>, as restarts are required when the bundle with native files change.
+
+This section details alternatives for using native code in the container.
+
+<h3>OSGi bundles containing native code</h3>
+
+<p>OSGi jars may contain .so files, which will can be loaded in the standard way from Java
+code in the bundle. Note that since only one instance of an .so can be loaded at any time,
```

Comment:
Minor but interesting: Should we read '.so' as "ess oh" or "dot ess oh"?  If the former, then 'an' is correct'. Otherwise it should of course be 'a'.

Reply:
Imho it's "ess of".